### ISSUE TITLE
Custom input prop boolean shorthand not working

### ISSUE THREAD

**Issue by benlind** [formkit/formkit#753]
### Reproduction

https://formkit.link/40fd78f583d09a156cee0128be17eac1

### Describe the bug

If you have a custom input that accepts a boolean prop, Vue's [boolean casting feature](https://vuejs.org/guide/components/props.html#boolean-casting) does not work. See the reproduction playground.

If I use this:

```
  <FormKit
    :plugins="[ myPlugin ]"
    type="custom"
    some-boolean
  />
```

Then the expected value of `props.context.someBoolean` in the custom component is `true`. But the actual value is an empty string.

If I instead use the long-hand syntax:

```
  <FormKit
    :plugins="[ myPlugin ]"
    type="custom"
    :some-boolean="true"
  />
```

Then `props.context.someBoolean` is correctly a `true` boolean.

### Environment

• OS: Web (MacOS)
• Browser Chrome
• Version 0.17

**Comment by justin-schroeder** [comment]
This is true — unfortunately we currently don’t have a great way to make this work "exactly" like vue, however in our own internal inputs we use the `undefine` utility which converts the prop to either `true` or `undefined` — in most cases `undefined` is actually preferable since it prevents an attribute by the same name from rendering if directly assigned.

For example: https://formkit.link/5e2b7df4f9b28dbf95910440c393884a

**Comment by benlind** [comment]
`undefined` works for me, thanks! It is indeed an awkward situation, to have to document this behavior in my codebase so future plugin devs know to use that utility. I'm tempted to ask you to make this the default. :)
